cmd/attest.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+)
+
+var keyType = "default"
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func(rw io.ReadWriter) (*client.Key, error){
+	"default": {
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gce": {
+		tpm2.AlgRSA: client.GceAttestationKeyRSA,
+		tpm2.AlgECC: client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest <default | gce>",
+	Short: "Create a remote attestation protobuf",
+	Long: `Create a remote attestation protobuf
+
+	The Attestation protobuf contains a Quote on all available PCR banks,
+	a way to validate the quote, and a TCG Event Log (Linux only).
+
+	The optional argument specifies the type of attestation key to create.
+	The command supports two attestation key types:
+	- DefaultAttestationKey
+	- GceAttestationKey.
+	The tool will generate the key if necessary and load the key for use.
+	The RSA key is used by default.
+	Use --algo to override the public key algorithm for the key.
+
+	--nonce attaches a nonce as extra data to the quote, guaranteeing a
+	fresh quote.
+	`,
+	Args: cobra.ExactValidArgs(1),
+	ValidArgs: func() []string {
+		// The keys from the hierarchyNames map are our valid arguments
+		aks := make([]string, len(attestationKeys))
+		for ak := range attestationKeys {
+			aks = append(aks, ak)
+		}
+		return aks
+	}(),
+	RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
args is not used, so the keyType will always default to "default", is this intended?

Reply:
nope. oversight on my part. thanks!